tree/treeplayer/inc/ROOT/TDFHistoModels.hxx

```diff
@@ -0,0 +1,94 @@
+// Author: Enrico Guiraud, Danilo Piparo CERN  09/2017
+
+/*************************************************************************
+ * Copyright (C) 1995-2017, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_TDFHISTOMODELS
+#define ROOT_TDFHISTOMODELS
+
+#include <TH1D.h>
+#include <TH2D.h>
+#include <TH3D.h>
+#include <memory>
+
+namespace ROOT {
+namespace Experimental {
+namespace TDF {
+
+class TH1DModel {
+public:
+   TString fName;
+   TString fTitle;
+   int fNbinsX;
+   double fXLow;
+   double fXUp;
+
+   TH1DModel() = delete;
+   TH1DModel(const TH1DModel &) = delete;
```

Comment:
would `default` instead of `delete` be fine here and in the corresponding ctors below?

Reply:
If we want to make the struct copiable, yes :) I decided in first approximation to turn off this possibility but being a mere collection of parameters, I guess that would not hurt...